arkouda/dataframe.py

```diff
@@ -1509,6 +1509,103 @@ def groupby(self, keys, use_series=True):
+
+        See Also
+        ________
+        ak.Series.isin
+
+        Notes
+        _____
+        - Pandas supports values being an iterable type. In arkouda, we replace this with pdarray
+        - Pandas supports ~ operations. Currently, ak.DataFrame does not support this.
+
+        Examples
+        ________
+        >>> df = ak.DataFrame({'col_A': ak.array([7, 3]), 'col_B':ak.array([1, 9])})
+        >>> df
+            col_A  col_B
+        0      7      1
+        1      3      9 (2 rows x 2 columns)
+
+        When `values` is a pdarray, check every value in the DataFrame to determine if it exists in values
+        >>> df.isin(ak.array([0, 1]))
+           col_A  col_B
+        0  False   True
+        1  False  False (2 rows x 2 columns)
+
+        When `values` is a dict, the values in the dict are passed to check the column indicated by the key
+        >>> df.isin({'col_A': ak.array([0, 3])})
+            col_A  col_B
+        0  False  False
+        1   True  False (2 rows x 2 columns)
+
+        When `values` is a Series, each column is checked if values is present positionally. This means that for `True`
+        to be returned, the indexes must be the same.
+        >>> i = ak.Index(ak.arange(2))
+        >>> s = ak.Series(data=[3, 9], index=i)
+        >>> df.isin(s)
+            col_A  col_B
+        0  False  False
+        1  False   True (2 rows x 2 columns)
+
+        When `values` is a DataFrame, the index and column must match. Note that 9 is not found because the column
+        name does not match.
+        >>> other_df = ak.DataFrame({'col_A':ak.array([7, 3]), 'col_C':ak.array([0, 9])})
+        >>> df.isin(other_df)
+            col_A  col_B
+        0   True  False
+        1   True  False (2 rows x 2 columns)
+        """
+        if isinstance(values, pdarray):
+            # flatten the DataFrame so single in1d can be used.
+            flat = concatenate(list(self.data.values()))
+            lens = array([self.data[col].size for col in self.columns])
+            segs = concatenate([cumsum(lens) - lens, array([flat.size])])
+            df_def = {col: in1d(flat, values)[segs[i]:segs[i + 1]] for i, col in enumerate(self.columns)}
+        elif isinstance(values, Dict):
+            # key is column name, val is the list of values to check
+            df_def = {col: (in1d(self.data[col], values[col]) if col in values.keys()
+                            else zeros(self.size, dtype=akbool)) for col in self.columns}
+        elif isinstance(values, DataFrame) or \
+                (isinstance(values, Series) and isinstance(values.index, Index)):
+            # create the dataframe with all false
+            df_def = {col: zeros(self.size, dtype=akbool) for col in self.columns}
+            # identify the indexes in both
+            rows_self = in1d(self.index.index, values.index.index)
+            rows_val = in1d(values.index.index, self.index.index)
```

Comment:
I'm not sure how performance compares between `in1d` vs `DataFrame.intersection` but we should be able to set the `unique` flag since we're dealing with indexes. So this is something to consider
```python
rows_self, rows_val = intersect(self.index.index, values.index.index, unique=True)
````

Reply:
I believe intersection is the better choice here. This has been updated.